components/ProductDetail/ProductHeart.tsx

```diff
@@ -12,37 +12,50 @@ interface ProductHeartProps {
   title: string;
 }
 
-// TODO: 유저가 찜을 했는지에 대한 여부에 따라 찜 하기와 찜 취소하기 분기 처리
 const ProductHeart = ({ productId, userId, title }: ProductHeartProps) => {
   const [isHeartProduct, setIsHeartHeartProduct] = useState(false);
   const toast = useToast();
   const router = useRouter();
 
-  const handleHeartClick = async () => {
-    if (userId === -1) {
-      toast(setToastInfo('top', '찜은 로그인 후 이용 가능합니다.', 'warning'));
-      router.push('/login');
+  useEffect(() => {
+    getProductHeartAuthUser();
+  }, []);
+
+  const getProductHeartAuthUser = async () => {
+    try {
+      const { heart } = (await heartAPI.getHeartAuthUser(productId)).data;
+      setIsHeartHeartProduct(heart);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const handleHeartClick = () => {
+    checkLoginAuthUser();
+    toggleHeartProduct();
+  };
+
+  const checkLoginAuthUser = () => {
+    if (userId !== -1) {
       return;
     }
 
-    if (!isHeartProduct) {
-      try {
-        await heartAPI.putHeart(userId, productId);
-        setIsHeartHeartProduct(true);
-        toast(setToastInfo('top', `${title} 상품을 찜했습니다!`, 'success'));
-      } catch (error) {
-        console.log(error);
-      }
-    } else {
-      try {
-        await heartAPI.deleteHeart(productId);
-        setIsHeartHeartProduct(false);
-        toast(
-          setToastInfo('top', `${title} 상품의 찜을 취소하였습니다.`, 'success')
-        );
-      } catch (error) {
-        console.log(error);
-      }
+    toast(setToastInfo('top', '찜은 로그인 후 이용 가능합니다.', 'warning'));
+    router.push('/login');
+  };
+
+  const toggleHeartProduct = async () => {
+    try {
+      await heartAPI.updateHeart(productId);
+      setIsHeartHeartProduct(!isHeartProduct);
+
+      isHeartProduct
+        ? toast(
+            setToastInfo('top', `${title}의 찜을 취소하였습니다.`, 'success')
+          )
+        : toast(setToastInfo('top', `${title}를 찜했습니다!`, 'success'));
```

Comment:
```suggestion
      const toastContent = isHeartProduct ? `${title}의 찜을 취소하였습니다.` :  `${title}를 찜했습니다!`;
      setToastInfo('top', toastContent, 'success');
```

로 정리해보았는데 어떻게 생각하시는지 궁금합니다!
변수명은 `toastContent` 또는 `toastText`정도로 생각해보았습니다👀

Reply:
너어무 좋아요! 반영하겠습니다!
자세히 봐주셔서 감사합니담!